src/spikeinterface/postprocessing/amplitude_scalings.py

```diff
@@ -0,0 +1,304 @@
+        unit_ind: unit_ids_to_channel_indices[unit_id] for unit_ind, unit_id in enumerate(unit_ids)
+    }
+
+    return worker_ctx
+
+
+def _amplitude_scalings_chunk(segment_index, start_frame, end_frame, worker_ctx):
+    # recover variables of the worker
+    spikes = worker_ctx["spikes"]
+    recording = worker_ctx["recording"]
+    all_templates = worker_ctx["all_templates"]
+    unit_inds_to_channel_indices = worker_ctx["unit_inds_to_channel_indices"]
+    nbefore = worker_ctx["nbefore"]
+    cut_out_before = worker_ctx["cut_out_before"]
+    cut_out_after = worker_ctx["cut_out_after"]
+    margin = worker_ctx["margin"]
+    return_scaled = worker_ctx["return_scaled"]
+
+    i0 = np.searchsorted(spikes["segment_ind"], segment_index)
+    i1 = np.searchsorted(spikes["segment_ind"], segment_index + 1)
+    spikes_in_segment = spikes[i0:i1]
+
+    i0 = np.searchsorted(spikes["segment_ind"], segment_index)
+    i1 = np.searchsorted(spikes["segment_ind"], segment_index + 1)
+    spikes_in_segment = spikes[i0:i1]
+
+    i0 = np.searchsorted(spikes_in_segment["sample_ind"], start_frame)
+    i1 = np.searchsorted(spikes_in_segment["sample_ind"], end_frame)
+
+    local_waveforms = []
+    templates = []
+
+    scalings = []
+
+    if i0 != i1:
+        local_spikes = spikes_in_segment[i0:i1]
+        traces_with_margin, left, right = get_chunk_with_margin(
+            recording._recording_segments[segment_index], start_frame, end_frame, channel_indices=None, margin=margin
+        )
+
+        # scale traces with margin to match scaling of templates
+        if return_scaled and recording.has_scaled():
+            gains = recording.get_property("gain_to_uV")
+            offsets = recording.get_property("offset_to_uV")
+            traces_with_margin = traces_with_margin.astype("float32") * gains + offsets
+
+        # get all waveforms
+        for spike in local_spikes:
+            unit_index = spike["unit_ind"]
+            sample_index = spike["sample_ind"]
+            sparse_indices = unit_inds_to_channel_indices[unit_index]
+            template = all_templates[unit_index][:, sparse_indices]
+            template = template[nbefore - cut_out_before : nbefore + cut_out_after]
+            sample_centered = sample_index - start_frame
+            cut_out_start = left + sample_centered - cut_out_before
+            cut_out_end = left + sample_centered + cut_out_after
+            if sample_index - cut_out_before < 0:
+                local_waveform = traces_with_margin[:cut_out_end, sparse_indices]
+                template = template[cut_out_before - sample_index :]
+            elif sample_index + cut_out_after > end_frame + right:
+                local_waveform = traces_with_margin[cut_out_start:, sparse_indices]
+                template = template[: -(sample_index + cut_out_after - end_frame)]
+            else:
+                local_waveform = traces_with_margin[cut_out_start:cut_out_end, sparse_indices]
```

Comment:
why this when we have margin ?

Reply:
This is a smaller cut out to get a local waveform

Reply:
yes but the margin ensure always the correct lenght no ?
When you get_chunk_with_margin(add_zeros=True)